src/main/java/at/hannibal2/skyhanni/test/TestCopyChestData.kt

```diff
@@ -0,0 +1,80 @@
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object TestCopyChestData {
+    private val config get() = SkyHanniMod.feature.dev.debug
+    private val mc = Minecraft.getMinecraft()
+    @SubscribeEvent
+    fun onKeybind(event: GuiScreenEvent.KeyboardInputEvent.Post) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!config.copyPlayerInventory.isKeyHeld() && !config.copyEntireChest.isKeyHeld() && !config.copyChestName.isKeyHeld()) return
+        if (event.gui !is GuiContainer) return
+        if (config.copyEntireChest.isKeyHeld() && event.gui is GuiChest) {
+            copySlotsData(chest = (event.gui as GuiChest).inventorySlots.inventorySlots)
+            return
+        } else if (config.copyPlayerInventory.isKeyHeld()) {
+            copySlotsData(inventory = if (config.includeArmor) mc.thePlayer.inventory.mainInventory + mc.thePlayer.inventory.armorInventory else mc.thePlayer.inventory.mainInventory)
+            return
+        } else if (InventoryUtils.openInventoryName().isNotEmpty()) {
+            OSUtils.copyToClipboard(InventoryUtils.openInventoryName())
+            LorenzUtils.chat("Chest name copied to clipboard.")
+            return
+        }
+    }
+    private fun copySlotsData(chest: List<Slot> = emptyList(), inventory: Array<ItemStack> = emptyArray()) {
+        val copyList = mutableListOf<String>("relevant config:", "includeNullSlots: ${config.includeNullSlots}", "includeUnnamedItems: ${config.includeUnnamedItems}", "includeArmor: ${config.includeArmor}", "")
+        val inventoryType: String = if (chest.isNotEmpty()) "Chest" else if (inventory.isNotEmpty()) "Inventory" else "Unknown"
+        if (chest.isNotEmpty()) {
+            copyList.addAll(listOf<String>("chest name: '${InventoryUtils.openInventoryName()}'", ""))
+            for (slot in chest) {
+                val stack = slot.stack
+                if (stack == null) {
+                    if (config.includeNullSlots) copyList.addAll(listOf("(there is nothing inside slot ${slot.slotIndex}; it is null)", "", ""))
+                    continue
+                }
+                if (stack in mc.thePlayer.inventory.mainInventory) break
+                if ((stack.displayName.isNotEmpty() && stack.displayName.isNotBlank()) || config.includeUnnamedItems) {
+                    copyList.add("slot index: '${slot.slotIndex}'")
+                    copyList.addAll(stack.getStackInfo())
+                }
+            }
+        } else if (inventory.isNotEmpty()) {
+            copyList.addAll(listOf<String>("your inventory is below.", "", "your hotbar:", ""))
+            for ((i, stack) in inventory.withIndex()) {
+                if (i - 1 == 8 && stack != null) copyList.addAll(listOf("note: hotbar ends here", "the rest of your inventory:", "", ""))
+                if (i - 1 == 35 && config.includeArmor && stack != null) copyList.addAll(listOf("note: the rest of your inventory data ends here", "your armor:", "", ""))
```

Comment:
Please use meaningful named constants instead:
```kt
val HOTBAR_SIZE = 9
val INVENTORY_SIZE = 27
```



```suggestion
                if (i == HOTBAR_SIZE) copyList.addAll(listOf("note: hotbar ends here", "the rest of your inventory:", "", ""))
                if (i == HOTBAR_SIZE + INVENTORY_SIZE && config.includeArmor) copyList.addAll(listOf("note: the rest of your inventory data ends here", "your armor:", "", ""))
```

I also removed the `stack != null` checks, since these would otherwise omit the note if the first slot of the respective inventory section is empty (irrespective of later slots being filled)

Reply:
done